File: LeetCode/May challenge/Day 13/Day_13.py
class Solution:
    def removeKdigits(self, num: str, k: int) -> str:
        stack = []
        
        for digit in num:
            while k > 0 and len(stack) > 0 and stack[-1] > digit:
                stack.pop()
                k -= 1
            stack.append(digit)
        
        while k > 0:
            stack.pop()
            k -= 1
        
        return "0" if len(stack) == 0 else str(int("".join(stack)))

File: LeetCode/May challenge/Day 13/Day_13.cpp
class Solution {
public:
    string removeKdigits(string num, int k) {
        if(num.length()<= k)
            return "0";
        string res;
        for(auto i: num){
            while(k && res.length() && res.back()>i)
            {
                res.pop_back();
                k--;
            }
            res.push_back(i);
        }
        res.resize(res.length()-k);
        int beg=0;
        while(beg< res.length()-1 && res[beg]== '0')
            beg++;
        
        return res.substr(beg, res.length() - beg + 1);
    }
};
